routetable/route_table.go

```diff
@@ -166,13 +184,29 @@ func (r *RouteTable) Apply() error {
 				r.logCxt.WithField("ifaceName", ifaceName).Debug(
 					"Resync: found calico-owned interface")
 				r.dirtyIfaces.Add(ifaceName)
+				r.onIfaceSeen(ifaceName)
 			}
 		}
+		// Clean up first-seen timestamps for old interfaces.
+		// Resyncs happen periodically, so the amount of memory leaked to old
+		// first seen timestamps is small.
+		for name, firstSeen := range r.ifaceNameToFirstSeen {
+			if r.dirtyIfaces.Contains(name) {
+				continue
```

Comment:
I got tripped up by relying on  `r.dirtyIfaces` containing all the interfaces here.  Could do with a comment or a slight refactor to use a local variable `allIfaces` then assign `allIfaces` to `r.dirtyIfaces`.  Yes, I know it's my code but it tripped me up! :-P